src/som/compiler/Parser.java

```diff
@@ -1444,6 +1448,38 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  /**
+   * Generates an implicit class nested onto the outer class.
+   *
+   * @return an expression that instantiates the implicit class
+   */
+  private ExpressionNode literalObject(final MethodBuilder builder)
+      throws ProgramDefinitionError {
+
+    // TODO: make note regarding the specification.
+
+    // Generate the implicit class's signature
+    String implicitClassSignature = "objL@" + lexer.getCurrentLineNumber() + "@"
+        + lexer.getCurrentColumn();
+    SourceSection source = getSource(getCoordinate());
+
+    // Setup the builder and "new" factory for the implicit class
+    MixinBuilder implicitClassBuilder = new MixinBuilder(builder,
+        AccessModifier.PUBLIC, symbolFor(implicitClassSignature), source,
+        structuralProbe);
+    MethodBuilder primaryFactory = implicitClassBuilder.getPrimaryFactoryMethodBuilder();
+    primaryFactory.addArgument("self", getEmptySource());
+    primaryFactory.setSignature(Symbols.NEW);
+    implicitClassBuilder.setupInitializerBasedOnPrimaryFactory(source);
+    inheritanceListAndOrBody(implicitClassBuilder);
+
+    // Generate the object literal node.
+    MixinDefinition mixinDef = implicitClassBuilder.assemble(source);
+    ObjectLiteralNode objectLiteral = new ObjectLiteralNode(mixinDef, builder.getSelfRead(source), source);
+    return SNodeFactory.createMessageSend(symbolFor("new"),
```

Comment:
hm, the structure is unintuitive to me. I think, I'd move the `#new` into the object literal node. So, the result of the object literal node is an object, and not a class.